src/SSTFrequencyLimitDetector.cpp

```diff
@@ -184,26 +194,18 @@ namespace geopm
     std::vector<std::pair<unsigned int, double> >
         SSTFrequencyLimitDetector::get_core_frequency_limits(unsigned int core_idx) const
     {
-#ifdef GEOPM_DEBUG
-        if (core_idx >= m_core_frequency_limits.size()) {
-            throw Exception("SSTFrequencyLimitDetector::" + std::string(__func__) +
-                                "Encountered invalid core index: " +
-                                std::to_string(core_idx),
-                            GEOPM_ERROR_INVALID,
-                            __FILE__, __LINE__);
-        }
-#endif
-        std::vector<std::map<unsigned int, double> > tradeoffs;
-
-        return m_core_frequency_limits[core_idx];
+        return m_core_frequency_limits.at(core_idx);
```

Comment:
I do like that this is cleaner. Question, though: what is our policy for managing exceptions? Is a class responsible for catching all internal exceptions and raising geopm::Exception (to provide a more standardized interface and hide internal implementation details) or do we allow underlying exceptions to propagate up?